src/rust/ide/enso-protocol/src/language_server.rs

```diff
@@ -115,99 +108,112 @@ pub enum FilesystemEventKind {
 
 /// Attributes of the file in the filesystem.
 #[derive(Clone,Debug,PartialEq,Eq,Hash)]
-#[derive(Serialize, Deserialize)]
+#[derive(Serialize,Deserialize)]
 #[serde(rename_all = "camelCase")]
 pub struct Attributes{
     /// When the file was created.
-    pub creation_time : UTCDateTime,
+    pub creation_time:UTCDateTime,
     /// When the file was last accessed.
-    pub last_access_time : UTCDateTime,
+    pub last_access_time:UTCDateTime,
     /// When the file was last modified.
-    pub last_modified_time : UTCDateTime,
+    pub last_modified_time:UTCDateTime,
     /// What kind of file is this.
-    pub kind : FileSystemObject,
+    pub kind:FileSystemObject,
     /// Size of the file in bytes.
     /// (size of files not being `RegularFile`s is unspecified).
-    pub byte_size : u64
+    pub byte_size:u64,
 }
 
 /// A representation of what kind of type a filesystem object can be.
 #[derive(Hash,Debug,Clone,PartialEq,Eq,Serialize,Deserialize)]
 #[serde(tag = "type")]
 pub enum FileSystemObject {
     /// Represents a directory.
-    Directory(Object),
+    Directory {
+        /// A name of the directory.
+        name:String,
+        /// A path to the directory.
+        path:Path,
+    },
```

Comment:
I'd rather put `#[allow(missing_docs)]` before enum, and remove the docs for all fields here (perhaps except the `target`).

Reply:
done